### ISSUE TITLE
bad output and pnsr when test_only

### ISSUE THREAD

**Issue by YDDDDG** [thstkdgus35/EDSR-PyTorch#290]
Hi,
There is a confusing problem.
When I train the model, the output psnr between the training process seems to be correct(around 30). however, when I use 'test_only', the psnr is only around 10. So I output the SR images and they are obviously wrong.

Here are the Set5 outputs 

![1](https://user-images.githubusercontent.com/33283950/100749362-88e0da80-341f-11eb-9d50-05d77de87efd.png)
![2](https://user-images.githubusercontent.com/33283950/100749442-a0b85e80-341f-11eb-880b-cad535669e63.png)
![3](https://user-images.githubusercontent.com/33283950/100749455-a615a900-341f-11eb-9687-a8129cdc30cb.png)
![4](https://user-images.githubusercontent.com/33283950/100749467-a8780300-341f-11eb-8ae3-3d40141d2258.png)
![5](https://user-images.githubusercontent.com/33283950/100749479-ab72f380-341f-11eb-9711-020859724324.png)

Do you have any idea about this strange output?

I look forward to and thank you for your reply！

Have a good day! ^_^

**Comment by xiaoyi-st** [comment]
> 嗨，
> 有一个令人困惑的问题。
> 当我训练模型时，训练过程之间的输出psnr似乎是正确的（大约30）。但是，当我使用“ test_only”时，psnr仅为10左右。因此，我输出SR图像，它们显然是错误的。
> 
> 这是Set5的输出
> 
> ![1个](https://user-images.githubusercontent.com/33283950/100749362-88e0da80-341f-11eb-9d50-05d77de87efd.png)
> ![2](https://user-images.githubusercontent.com/33283950/100749442-a0b85e80-341f-11eb-880b-cad535669e63.png)
> ![3](https://user-images.githubusercontent.com/33283950/100749455-a615a900-341f-11eb-9687-a8129cdc30cb.png)
> ![4](https://user-images.githubusercontent.com/33283950/100749467-a8780300-341f-11eb-8ae3-3d40141d2258.png)
> ![5](https://user-images.githubusercontent.com/33283950/100749479-ab72f380-341f-11eb-9711-020859724324.png)
> 
> 您对这种奇怪的输出有任何想法吗？
> 
> 期待与您的回复！
> 
> 祝你有美好的一天！^ _ ^

你好，这个问题您解决了吗

**Comment by YDDDDG** [comment]
> > 嗨，
> > 有一个令人困惑的问题。
> > 当我训练模型时，训练过程之间的输出psnr似乎是正确的（大约30）。但是，当我使用“ test_only”时，psnr仅为10左右。因此，我输出SR图像，它们显然是错误的。
> > 这是Set5的输出
> > ![1个](https://user-images.githubusercontent.com/33283950/100749362-88e0da80-341f-11eb-9d50-05d77de87efd.png)
> > ![2](https://user-images.githubusercontent.com/33283950/100749442-a0b85e80-341f-11eb-880b-cad535669e63.png)
> > ![3](https://user-images.githubusercontent.com/33283950/100749455-a615a900-341f-11eb-9687-a8129cdc30cb.png)
> > ![4](https://user-images.githubusercontent.com/33283950/100749467-a8780300-341f-11eb-8ae3-3d40141d2258.png)
> > ![5](https://user-images.githubusercontent.com/33283950/100749479-ab72f380-341f-11eb-9711-020859724324.png)
> > 您对这种奇怪的输出有任何想法吗？
> > 期待与您的回复！
> > 祝你有美好的一天！^ _ ^
> 
> 你好，这个问题您解决了吗

No, but I guess there's something wrong when loading the pretrained model

**Comment by Doreenqiuyue** [comment]
我也遇到了同样的问题，你们解决了吗??

**Comment by zhangqizky** [comment]
我也是，这是啥情况啊？有时候输出的是那种feature-like的图像，真的是醉了。。。。

**Comment by zhangqizky** [comment]
> 我也遇到了同样的问题，你们解决了吗??

我知道原因了。EDSR_2.pt这个模型，不知道作者使用的哪些超参数训练的，指的不是res_blocks和n_feats，是res_scale这一项，这一项的默认参数是1。我加载自己的模型，训练和测试的res_scale都是0.1，结果就对了。

**Comment by Doreenqiuyue** [comment]
> > 我也遇到了同样的问题，你们解决了吗??
> 
> 我知道原因了。EDSR_2.pt这个模型，不知道作者使用的哪些超参数训练的，指的不是res_blocks和n_feats，是res_scale这一项，这一项的默认参数是1。我加载自己的模型，训练和测试的res_scale都是0.1，结果就对了。

谢谢解答! 想问一下res_scale的作用是什么，一直都不太明白

**Comment by 1187697147** [comment]
@zhangqizky
 啊这，碰到buaa的同学

**Comment by xiaoyi-st** [comment]
？




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Egqawkq"<notifications@github.com&gt;; 
发送时间: 2020年12月20日(星期天) 晚上10:40
收件人: "thstkdgus35/EDSR-PyTorch"<EDSR-PyTorch@noreply.github.com&gt;; 
抄送: "R"<1120922880@qq.com&gt;; "Comment"<comment@noreply.github.com&gt;; 
主题: Re: [thstkdgus35/EDSR-PyTorch] bad output and pnsr when test_only (#290)





 
@zhangqizky
 啊这，碰到buaa的同学

**Comment by zhangqizky** [comment]
> @zhangqizky
> 啊这，碰到buaa的同学

哈哈，这个问题，好像还是不太对，也是醉了。

**Comment by xuelimin** [comment]
我出现这个原因（test）是因为加载的模型是*2的，但在命令行输入的scale是*4的，尺寸不匹配出图就是上面的亚子~~~

**Comment by shenhaiyoulan** [comment]
I meet all the model meet the same problem,do you solve them?

**Comment by Godkingbin** [comment]
have u solve the problem?

**Comment by zhangqizky** [comment]
您的邮件已收到，我会尽快回复，谢谢。张琦

**Comment by Liiiiaictx** [comment]
@xuelimin 大佬，你是怎么测试的？是用预训练的模型测试的吗还是自己训练的得到的？请问该怎么用作者训练的模型取测试基准数据集，需要设置那些参数？此外，我想问下你的环境设置是什么样的？期待你的回复

**Comment by xuelimin** [comment]
邮件已查收，我会尽快回复您！

**Comment by xiaoyi-st** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，邮件已收到

**Comment by Liiiiaictx** [comment]
@zhangqizky 大佬，你是怎么测试的？是用预训练的模型测试的吗还是自己训练的得到的？请问该怎么用作者训练的模型取测试基准数据集，需要设置那些参数？此外，我想问下你的环境设置是什么样的，如pytorch版本，cuda版本？我自己测试总是出错，希望可以得到您的帮助，期待你的回复

**Comment by anhuigzj** [comment]
你好，EDSR的代码我能运行，但是在验证阶段不计算PSNR，在trainer.py文件中‘for lr, hr, filename in tqdm(d, ncols=80)‘，该循环并没有执行，也就是验证阶段数据并没有加载进去，不清楚什么原因。在option.py文件中data_train和data_test参数分别是DIV2K和Set14

**Comment by anhuigzj** [comment]
> 您的邮件已收到，我会尽快回复，谢谢。张琦

你好，EDSR的代码我能运行，但是在验证阶段不计算PSNR，在trainer.py文件中‘for lr, hr, filename in tqdm(d, ncols=80)‘，该循环并没有执行，也就是验证阶段数据并没有加载进去，不清楚什么原因。在option.py文件中data_train和data_test参数分别是DIV2K和Set14

**Comment by zhangqizky** [comment]
您的邮件已收到，我会尽快回复，谢谢。张琦

**Comment by xuelimin** [comment]
邮件已查收，我会尽快回复您！

**Comment by xiaoyi-st** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，邮件已收到

**Comment by anhuigzj** [comment]
> 邮件已查收，我会尽快回复您！

你好，EDSR的代码我能运行，但是在验证阶段不计算PSNR，在trainer.py文件中‘for lr, hr, filename in tqdm(d, ncols=80)‘，该循环并没有执行，也就是验证阶段数据并没有加载进去，不清楚什么原因。在option.py文件中data_train和data_test参数分别是DIV2K和Set14
